ambari-server/src/main/java/org/apache/ambari/server/state/cluster/ClusterImpl.java

```diff
@@ -2199,6 +2201,18 @@ public Long getNextConfigVersion(String type) {
         Enum<?> currentState = e.getCurrentState();
         Enum<?> failedEvent = e.getEvent();
 
+        if (serviceComponent != null && serviceComponentHost != null
+            && (currentState == State.STARTED || currentState == State.STARTING)
+            && failedEvent == ServiceComponentHostEventType.HOST_SVCCOMP_INSTALL
+            && serviceComponentName.equals("NODEMANAGER") && serviceComponentHost
```

Comment:
The logic should be generalized without special treatment for any pre-defined component

Reply:
Ok then removing NM from this logic should be good enough for us.

Reply:
@hapylestat - I have removed the treatment for NM specifically but overall retained the check with the state and the failedEvent as is.

Reply:
@hapylestat 
Can you have a look at the patch now?

Reply:
@hapylestat 
Ping for reviews.